src/pudl_archiver/archivers/ferc/xbrl.py

```diff
@@ -312,6 +354,15 @@ async def archive_year(
             filename = f"{filing.title}_form{filing.ferc_formname.as_int()}_{filing.ferc_period}_{round(filing.published_parsed.timestamp())}.xbrl".replace(
                 " ", "_"
             )
+            filing_name = f"{filing.title}{filing.ferc_period}"
+            if filing_name not in metadata:
```

Comment:
non-blocking: should we use a defaultdict here?

Reply:
Yes! For some reason I *always* forget that `defaultdict` exists even though it's super useful!